lib/dao.js

```diff
@@ -2638,8 +2638,12 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
   assert(typeof cb === 'function', 'The cb argument must be a function');
 
   var connector = this.getConnector();
-  assert(typeof connector.replaceById === 'function',
-          'replaceById() must be implemented by the connector');
+
+  if (typeof connector.replaceById !== 'function') {
+    var err =  new Error('there is no support for replaceById() in the connector.' +
+      'Connector needs to add support for `replaceById()`');
```

Comment:
Let's rephrase this error message. First of all, start with capital letter (sentence case). Secondly, make the error message actionable - provide clear call to action for the reader.

Last but not least, the error message must be globalized!

```
var err = new Error(g.f(
  'The connector %s does not support {{replaceById}} operation. This is not a bug in LoopBack. ' +
    'Please contact the authors of the connector, preferable via GitHub issues.',
  connector.name));
```